aodntools/timeseries_products/velocity_aggregated_timeseries.py

```diff
@@ -0,0 +1,373 @@
+import os
+import sys
+import tempfile
+import shutil
+import netCDF4 as nc4
+import numpy as np
+import json
+from datetime import datetime
+import argparse
+from pkg_resources import resource_filename
+from aodntools import __version__
+
+import xarray as xr
+
+import aggregated_timeseries as TStools
+
+TEMPLATE_JSON = resource_filename(__name__, 'velocity_aggregated_timeseries_template.json')
+
+
+def sort_files(file_list, input_dir=""):
+    """
+    sort list of files according to deployment date
+    :param file_list: List of files to sort
+    :return: sorted list of files
+    """
+
+    time_start = []
+    for file in file_list:
+        with nc4.Dataset(os.path.join(input_dir, file), 'r') as ds:
+            time_start.append(np.datetime64(ds.time_deployment_start))
+    tuples = sorted(zip(time_start, files_to_agg))
+    return [t[1] for t in tuples]
+
+
+def check_file(nc, site_code):
+    """
+    Return list of errors found in the file:
+    Variables of interest are present
+    TIME, DEPTH, LATITUDE, LONGITUDE,  is present
+    NOMINAL_DEPTH is not present as variable or attribute
+    file_version is not FV01
+    if LATITUDE or LONIGITUDE dimension has length >1
+
+    :param nc: xarray dataset
+    :param site_code: code of the mooring site
+    :return: dictionary with the file name and list of failed tests
+    """
+
+    attributes = list(nc.attrs)
+    variables = list(nc.variables)
+    allowed_dimensions = ['TIME', 'LATITUDE', 'LONGITUDE', 'HEIGHT_ABOVE_SENSOR']
+    required_variables = ['UCUR', 'VCUR', 'WCUR']
```

Comment:
I've removed the WCUR check from the file_check function. I think we can leave the WCUR handle in the code even if WCUR are expected to be present. It will be safe in case the variable disappears in the future. It does not harm.

Reply:
:+1: